src/main/webapp/app/exam/participate/exam-navigation-sidebar/exam-navigation-sidebar.component.ts

```diff
@@ -0,0 +1,254 @@
+})
+export class ExamNavigationSidebarComponent implements OnDestroy, OnInit {
+    @Input() sidebarData: SidebarData;
+    @Input() exercises: Exercise[] = [];
+    @Input() exerciseIndex = 0;
+    @Input() overviewPageOpen: boolean;
+    @Input() examSessions?: ExamSession[] = [];
+    @Input() examTimeLineView = false;
+    @Input() isTestRun = 0;
+    @Output() onPageChanged = new EventEmitter<{
+        overViewChange: boolean;
+        exercise?: Exercise;
+        forceSave: boolean;
+        submission?: ProgrammingSubmission | SubmissionVersion | FileUploadSubmission;
+    }>();
+
+    static itemsVisiblePerSideDefault = 4;
+    itemsVisiblePerSide = ExamNavigationSidebarComponent.itemsVisiblePerSideDefault;
+
+    subscriptionToLiveExamExerciseUpdates: Subscription;
+
+    // Icons
+    icon: IconProp;
+    overviewIcon: IconProp = faFileLines;
+    faChevronRight = faChevronRight;
+    readonly ExerciseButtonStatus = ExerciseButtonStatus;
+
+    profileSubscription?: Subscription;
+    isProduction = true;
+    isTestServer = false;
+    isCollapsed: boolean = false;
+    exerciseId: string;
+    numberOfSavedExercises: number = 0;
+
+    constructor(
+        private profileService: ProfileService,
+        private sidebarEventService: SidebarEventService,
+        private layoutService: LayoutService,
+        private examParticipationService: ExamParticipationService,
+        private examExerciseUpdateService: ExamExerciseUpdateService,
+        private repositoryService: CodeEditorRepositoryService,
+        private conflictService: CodeEditorConflictStateService,
+    ) {}
+
+    ngOnInit(): void {
+        this.profileSubscription = this.profileService.getProfileInfo()?.subscribe((profileInfo) => {
+            this.isProduction = profileInfo?.inProduction;
+            this.isTestServer = profileInfo?.testServer ?? false;
+        });
+
+        if (!this.examTimeLineView) {
+            this.subscriptionToLiveExamExerciseUpdates = this.examExerciseUpdateService.currentExerciseIdForNavigation.subscribe((exerciseIdToNavigateTo) => {
+                // another exercise will only be displayed if the student clicks on the corresponding pop-up notification
+                this.changeExerciseById(exerciseIdToNavigateTo);
+            });
+        }
+
+        this.layoutService.subscribeToLayoutChanges().subscribe(() => {
+            // You will have all matched breakpoints in observerResponse
+            if (this.layoutService.isBreakpointActive(CustomBreakpointNames.extraLarge)) {
+                this.itemsVisiblePerSide = ExamNavigationSidebarComponent.itemsVisiblePerSideDefault;
+            } else if (this.layoutService.isBreakpointActive(CustomBreakpointNames.large)) {
+                this.itemsVisiblePerSide = 3;
+            } else if (this.layoutService.isBreakpointActive(CustomBreakpointNames.medium)) {
+                this.itemsVisiblePerSide = 1;
+            } else {
+                this.itemsVisiblePerSide = 0;
```

Comment:
Could you please define these magic numbers as constants and give them descriptive names? E.g. it is not clear for me right now which case "0" describes without knowing the UI (does it mean nothing is displayed at the left side?)

Reply:
This part was copied from `exam-navigation-bar.component.ts`, but I think it is not necessary for sidebar implementation anymore. I will remove this part